server/game/gameSystems/ResourceCardSystem.ts

```diff
@@ -28,15 +29,21 @@ export class ResourceCardSystem<TContext extends AbilityContext = AbilityContext
         // TODO: remove this completely if determinmed we don't need card snapshots
         // event.cardStateWhenMoved = card.createSnapshot();
         const properties = this.generatePropertiesFromContext(context, additionalProperties) as IResourceCardProperties;
-        //TODO: Is there a better/cleaner way to handle one or multiple cards here?
-        const cards = [].concat(properties.target);
-        cards.forEach((card) => {
-            const player = properties.changePlayer && card.controller.opponent ? card.controller.opponent : card.controller;
-            player.moveCard(card, Location.Resource);
-            if (properties.readyResource) {
-                context.game.openEventWindow(ready({ target: card }).generateEvent(context.source, context));
-            }
-        });
+        // TODO: If we ever need to resource multiple cards at once, this will need an update
+        const card = Array.isArray(properties.target) ? properties.target[0] as Card : properties.target as Card;
```

Comment:
Can we override `generatePropertiesFromContext` and add a contract check there that `properties.target.length === 1` if it is an array, that way if someone tries it without realizing they'll get notified and also we don't have to put that check everywhere in here this gets used.